gotool/gotool_test.go

```diff
@@ -0,0 +1,74 @@
+package gotool
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type TestCommander struct {
+	ExecuteCommandFn func(command string, args ...string) (string, error)
```

Comment:
While this looks a pretty good custom solution, I'd urge you to take a look at [testify](https://github.com/stretchr/testify), which provides a standard way of mocking objects.
For an example usage, take a look here: https://github.com/bitrise-io/beitris/pull/38/files#diff-8d6851f3ab5c8373ede6234121198af8